File: regalia-moshi/src/test/java/com/commit451/regalia/moshi/Cat.java
package com.commit451.regalia.moshi;

import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;

public class Cat extends RealmObject {

    String name;
    RealmList<Cat> friends;

    public List<Cat> getFriends() {
        return friends;
    }
}

File: regalia-gson/src/test/java/com/commit451/regalia/gson/Cat.kt
package com.commit451.regalia.gson

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Cat : RealmObject() {
    @PrimaryKey
    var name: String? = null
    var friends: RealmList<Cat>? = null
}